### PR TITLE
Updated to SCP format version 1.6.

### PR DESCRIPTION
**Pull request by claunia**

### CODE REVIEW

**Comment by claunia** [comment]
And added correct checksum calculation.

File: libdisk/container/scp.c [hunk 1]
```diff
@@ -231,6 +231,28 @@ static void scp_close(struct disk *d)
 
     lseek(d->fd, sizeof(dhdr), SEEK_SET);
     write_exact(d->fd, th_offs, di->nr_tracks * sizeof(uint32_t));
+
+    size_t filesize = (size_t)lseek(d->fd, 0, SEEK_END);
+    unsigned char* buffer = malloc(filesize);
```
Comment (simonowen) [inline libdisk/container/scp.c]: missing a free(buffer) below?

File: scp/scp_dump.c [hunk 2]
```diff
@@ -256,5 +256,27 @@ int main(int argc, char **argv)
     lseek(fd, sizeof(dhdr), SEEK_SET);
     write_exact(fd, th_offs, SCP_MAX_TRACKS * sizeof(uint32_t));
 
+    size_t filesize = (size_t)lseek(fd, 0, SEEK_END);
+    unsigned char* buffer = malloc(filesize);
```
Comment (simonowen) [inline scp/scp_dump.c]: here too?

Comment (claunia) [reply scp/scp_dump.c]: True!

**Comment by keirf** [comment]
I reworked the patch series and committed to master.

**Pull request closed without merging**